src/lib/Telemetry/telemetry.cpp

```diff
@@ -204,120 +204,85 @@ bool Telemetry::RXhandleUARTin(uint8_t data)
     return true;
 }
 
-bool Telemetry::AppendTelemetryPackage(uint8_t *package)
+void Telemetry::AppendTelemetryPackage(uint8_t *package)
 {
-    const crsf_header_t *header = (crsf_header_t *) package;
+    const crsf_ext_header_t *header = (crsf_ext_header_t *) package;
 
-    if (header->type == CRSF_FRAMETYPE_COMMAND && package[3] == 'b' && package[4] == 'l')
+    //handle non-OTA
+    if (header->type == CRSF_FRAMETYPE_COMMAND && package[3] == 'b' && package[4] == 'l') // note: this should be an extended csrf packet, not a standard crsf packet
     {
         callBootloader = true;
-        return true;
+        return;
     }
-    if (header->type == CRSF_FRAMETYPE_COMMAND && package[3] == 'b' && package[4] == 'd')
+    if (header->type == CRSF_FRAMETYPE_COMMAND && package[3] == 'b' && package[4] == 'd') // note: this should be an extended csrf packet, not a standard crsf packet
     {
         callEnterBind = true;
-        return true;
+        return;
     }
-    if (header->type == CRSF_FRAMETYPE_COMMAND && package[3] == 'm' && package[4] == 'm')
+    if (header->type == CRSF_FRAMETYPE_COMMAND && package[3] == 'm' && package[4] == 'm') // note: this should be an extended csrf packet, not a standard crsf packet
     {
         callUpdateModelMatch = true;
         modelMatchId = package[5];
-        return true;
+        return;
     }
-    if (header->type == CRSF_FRAMETYPE_DEVICE_PING && package[CRSF_TELEMETRY_TYPE_INDEX + 1] == CRSF_ADDRESS_CRSF_RECEIVER)
+    if (header->type == CRSF_FRAMETYPE_DEVICE_PING && header->dest_addr == CRSF_ADDRESS_CRSF_RECEIVER)
```

Comment:
We need the conditon (header->type == CRSF_FRAMETYPE_DEVICE_PING && (header->dest_addr == 0 || header->dest_addr == CRSF_ADDRESS_CRSF_RECEIVER)) since it's also possible to send a ping to all devices